frontends/web/src/routes/settings/manage-accounts.tsx

```diff
@@ -154,6 +154,23 @@ class ManageAccounts extends Component<Props, State> {
     }
   };
 
+  private handleSetWatch = async (accountCode: string, watch: boolean) => {
+    if (watch === false) {
+      this.setState({ storedAccountCode: accountCode });
+      // attempting to connect to its keystore
+      // should open ConfirmDialogDisableWatchonly if keystore isn't loaded
+      const hasConnectedKeystore = await accountAPI.connectKeystore(accountCode);
```

Comment:
Why ask the user to connect the device just to disable watch-only? why not just ask them for confirmation with the appropriate warning directly?